### PR TITLE
New specs and parsers for scsi_mod, lpfc driver and qla2xxx driver ma…

### PR DESCRIPTION
**Pull request by qinpingli**
…x luns

### All Pull Requests:

Check all that apply:

* [X] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

These specs and parsers are used to detect linux reports lun<nnn> has a LUN larger than allowed by the host adapter issue.  Detailed info is in the CEECBA-5708.

### CODE REVIEW

File: insights/parsers/sys_module.py [hunk 1]
```diff
@@ -101,3 +131,45 @@ class VHostNetZeroCopyTx(XModUseBlkMq):
...
     pass
+
+
+@parser(Specs.lpfc_max_luns)
+class LpfcMaxLUNs(MaxLUNs):
+    """
+    This file `/sys/module/lpfc/parameters/lpfc_max_luns` shows the max LUN number
+    supported by lpfc driver.
+
+    Examples::
```
Comment (bfahr) [inline insights/parsers/sys_module.py:142]: `Examples:` should only have 1 colon.  See the [recommended style guide](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for more information.

File: insights/parsers/sys_module.py [hunk 2]
```diff
@@ -101,3 +131,45 @@ class VHostNetZeroCopyTx(XModUseBlkMq):
...
+    pass
+
+
+@parser(Specs.ql2xmaxlun)
+class Ql2xMaxLUN(MaxLUNs):
+    """
+    This file `/sys/module/qla2xxx/parameters/ql2xmaxlun` shows the max LUN number
+    supported by qla2xxxx driver.
+
+    Examples::
```
Comment (bfahr) [inline insights/parsers/sys_module.py:156]: Same comment as above.

File: insights/parsers/sys_module.py [hunk 3]
```diff
@@ -101,3 +131,45 @@ class VHostNetZeroCopyTx(XModUseBlkMq):
...
+    pass
+
+
+@parser(Specs.scsi_mod_max_report_luns)
+class SCSIModMaxReportLUNs(MaxLUNs):
+    """
+    This file `/sys/module/scsi_mod/parameters/max_report_luns` shows the max LUN number
+    supported by OS.
+
+    Examples::
```
Comment (bfahr) [inline insights/parsers/sys_module.py:170]: Same comment as above.

File: insights/parsers/sys_module.py [hunk 4]
```diff
@@ -55,6 +61,30 @@ def is_on(self):
...
+
+    Sample Content::
+
+        512
+
+    Raises:
+        SkipException: When content is empty or no parse-able content.
+
+    Attributes:
+        val(str): Raw data of the content.
```
Comment (bfahr) [inline insights/parsers/sys_module.py:77]: @qinpingli the type for this attribute should be `int`.

**Review by bfahr — CHANGES_REQUESTED** [review]

**Comment by qinpingli** [comment]
@bfahr Thanks for the review! Created a commit to fix them, please help take a look again.

**Review by vdymna — APPROVED** [review]
This looks good to me.

**Pull request merged**